accio-base/src/main/java/io/accio/base/config/ConfigManager.java

```diff
@@ -0,0 +1,345 @@
+import static io.accio.base.client.duckdb.DuckDBConfig.DUCKDB_HOME_DIRECTORY;
+import static io.accio.base.client.duckdb.DuckDBConfig.DUCKDB_MAX_CACHE_QUERY_TIMEOUT;
+import static io.accio.base.client.duckdb.DuckDBConfig.DUCKDB_MAX_CONCURRENT_QUERIES;
+import static io.accio.base.client.duckdb.DuckDBConfig.DUCKDB_MAX_CONCURRENT_TASKS;
+import static io.accio.base.client.duckdb.DuckDBConfig.DUCKDB_MEMORY_LIMIT;
+import static io.accio.base.client.duckdb.DuckDBConfig.DUCKDB_TEMP_DIRECTORY;
+import static io.accio.base.client.duckdb.DuckdbS3StyleStorageConfig.DUCKDB_STORAGE_ACCESS_KEY;
+import static io.accio.base.client.duckdb.DuckdbS3StyleStorageConfig.DUCKDB_STORAGE_ENDPOINT;
+import static io.accio.base.client.duckdb.DuckdbS3StyleStorageConfig.DUCKDB_STORAGE_REGION;
+import static io.accio.base.client.duckdb.DuckdbS3StyleStorageConfig.DUCKDB_STORAGE_SECRET_KEY;
+import static io.accio.base.client.duckdb.DuckdbS3StyleStorageConfig.DUCKDB_STORAGE_URL_STYLE;
+import static io.accio.base.config.AccioConfig.ACCIO_DATASOURCE_TYPE;
+import static io.accio.base.config.AccioConfig.ACCIO_DIRECTORY;
+import static io.accio.base.config.AccioConfig.ACCIO_ENABLE_DYNAMIC_FIELDS;
+import static io.accio.base.config.AccioConfig.ACCIO_FILE;
+import static io.accio.base.config.BigQueryConfig.BIGQUERY_BUCKET_NAME;
+import static io.accio.base.config.BigQueryConfig.BIGQUERY_CRENDITALS_FILE;
+import static io.accio.base.config.BigQueryConfig.BIGQUERY_CRENDITALS_KEY;
+import static io.accio.base.config.BigQueryConfig.BIGQUERY_LOCATION;
+import static io.accio.base.config.BigQueryConfig.BIGQUERY_METADATA_SCHEMA_PREFIX;
+import static io.accio.base.config.BigQueryConfig.BIGQUERY_PARENT_PROJECT_ID;
+import static io.accio.base.config.BigQueryConfig.BIGQUERY_PROJECT_ID;
+import static io.accio.base.config.PostgresConfig.POSTGRES_JDBC_URL;
+import static io.accio.base.config.PostgresConfig.POSTGRES_PASSWORD;
+import static io.accio.base.config.PostgresConfig.POSTRES_USER;
+import static io.accio.base.config.PostgresWireProtocolConfig.PG_WIRE_PROTOCOL_AUTH_FILE;
+import static io.accio.base.config.PostgresWireProtocolConfig.PG_WIRE_PROTOCOL_NETTY_THREAD_COUNT;
+import static io.accio.base.config.PostgresWireProtocolConfig.PG_WIRE_PROTOCOL_PORT;
+import static io.accio.base.config.PostgresWireProtocolConfig.PG_WIRE_PROTOCOL_SSL_ENABLED;
+import static io.accio.base.metadata.StandardErrorCode.NOT_FOUND;
+import static java.util.stream.Collectors.toList;
+
+public class ConfigManager
+{
+    private Optional<AccioConfig> accioConfig;
+    private Optional<PostgresConfig> postgresConfig;
+    private Optional<BigQueryConfig> bigQueryConfig;
+    private Optional<DuckDBConfig> duckDBConfig;
+    private Optional<PostgresWireProtocolConfig> postgresWireProtocolConfig;
+    private Optional<DuckdbS3StyleStorageConfig> duckdbS3StyleStorageConfig;
+
+    private ConcurrentHashMap<String, String> configs = new ConcurrentHashMap<>();
```

Comment:
I thought we will just update the PostgresConfig and BigQueryConfig, etc. Why we specially save the config cache?

Reply:
Using Map to handle all configs, it can help us to manage the configurations efficiently. By cache all config, we don't need to create a new config instance when we want to get the latest configuration everytime.